### DISCUSSION TITLE
send ip in headers (sveltekit server side)

### DISCUSSION THREAD

**Discussion by lIMonkieIl** [pocketbase/pocketbase#5889] [Q&A]
hello community,

so i have a sveltekit app that works fine the only issue i am having is getting the users ip in the logs so i can  then also use the rate limiting so this is what i have done and  confirmed works so far.

so connecting to pocketbase via client side i successfully see the users ip and user agent and such

when conencting via serverside i only see user-agent as node and the ccontainer ip  so what i did was forward headers in the pb.beforSend like so 

```typescript
export const initPocketbase: Handle = async ({ event, resolve }) => {
  event.locals.pb = new PocketBase(env.POCKETBASE_BASE_URL);
  event.locals.pb.beforeSend = function (url, options) {
    const headers = Object.fromEntries(event.request.headers.entries());
    options.headers = Object.assign({}, options.headers, headers);
    return { url, options };
  };
  const response = await resolve(event);

  return response;
};
```

this successfully set the user-agent and referer but its not setting the ip

i am behind a proxy and have set the correct header in the pocketbase settings.

any help wuld be amazing, thanks

**Comment by ganigeorgiev** [comment]
I'm confused and not sure that I understand your setup.

How are you communicating with PocketBase - only client-side, only server-side or mixed (both)?

You've mentioned that you are behind a proxy, but if you are communicating with PocketBase server-side (or mixed) from SvelteKit you have in practice 2 proxies (client -> proxy -> node -> PocketBase).

Are you sure that you are forwarding the headers correctly? What is `Object.fromEntries(event.request.headers.entries())` returning?

**Reply by lIMonkieIl** [reply]
Hello,

I'm doing it as hybrid and I understand I need to keep both auth states in sync.
And the issue only happens server side on the client side all correct data is sent

If I print the event.request.headers to console

I will see the correct data ie I see the users  ip under cf-connecting-ip and such 

And also when I forward the headers I do see the correct user-agent and refere but the ip I don't see 

The ip that seems to get used is under x-real-ip and x-forwarded-for they both are my container ip

**Reply by ganigeorgiev** [reply]
Without anything more specific I don't think I can help you.

By saying _"I see the users ip under cf-connecting-ip and such"_ doesn't really narrow the issue.

If you see the user IP under the `CF-Connecting-IP` is this the header that you've configured in the PocketBase settings?

**Reply by ganigeorgiev** [reply]
Just to make sure that we are on the same page, because you've mentioned rate limiting I assume that you use PocketBase v0.23.0-rc, right?

**Reply by lIMonkieIl** [reply]
I'm happy to share any code and information needed.

Yes cf-connecting-ip is the Header I have set in pocketbase and I can confirm this is correct as logging in via the client side I get the correct data ie users ip and user-gent get set

**Reply by lIMonkieIl** [reply]
> PocketBase v0.23.0-rc

Yes sorry should have said

**Reply by ganigeorgiev** [reply]
I'm sorry I still don't understand what is your setup and how your code actually runs.

I hate this back-and-forth and I cannot help you further, but if you want someone else to better understand your issue and assist you:
- elaborate more on your setup, how your applications are deployed and how exactly did you configure your proxy
- inspect and provide the exact request flow that is getting triggered by a single client (aka. at every step which exact service is getting triggered and how, aka. with what address)
- create a **minimal** reproducible illustrating the problem.

**Reply by ganigeorgiev** [reply]
Also if you are connecting from the Node server-side to PocketBase VIA the Cloudflare proxied address and not directly, then I doubt that manually setting `CF-Connecting-IP` will work because I'd expect Cloudflare to overwrite the header with the actual connecting client (in your case the node server) to prevent spoofing.

**Reply by lIMonkieIl** [reply]
Thanks for replying and  helping,

I have managed to achieve what I needed.

By using traefik via a plugin to copy the cf-connecting-ip header to x-preppal-ip header if the x-preppal-ip does not exist on the request already.

Then in pocketbase I set the proxy header to x-preppal-ip and all seems to work
